compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -3209,6 +3209,10 @@ private void defineFieldsOfClassDef(BLangClassDefinition classDefinition, Symbol
 
         for (BLangSimpleVariable field : classDefinition.fields) {
             defineNode(field, typeDefEnv);
+            if (field.expr != null) {
+                field.symbol.flags |= Flags.OPTIONAL;
```

Comment:
I'm not exactly sure about this. We've used the `OPTIONAL` flag solely in record contexts. Not sure if it's a good idea to mix it in other contexts as well, in case it gets misconstrued.